spring/src/main/kotlin/de/codecentric/hikaku/converters/spring/extensions/ProducesSpringExtension.kt

```diff
@@ -63,7 +68,16 @@ private fun HandlerMethod.isResponseBodyAnnotationOnClass() = this.method
         ?.jvmErasure
         ?.findAnnotation<ResponseBody>() != null
 
-
 private fun HandlerMethod.isResponseBodyAnnotationOnFunction() = this.method
         .kotlinFunction
-        ?.findAnnotation<ResponseBody>() != null
\ No newline at end of file
+        ?.findAnnotation<ResponseBody>() != null
+
+private val javaxServletResponseClass: Class<*>? =
+        (try {
+            Class.forName("javax.servlet.http.HttpServletResponse")
+        } catch (ex: Throwable) {
+            null
+        })
+
+private fun HandlerMethod.hasHttpServletResponseParam() = this.methodParameters
+        .any { javaxServletResponseClass !== null && it.parameterType.isAssignableFrom(javaxServletResponseClass) }
```

Comment:
You get the Class using the full qualified class name here. Why not check the name directly in the any with that string like:
```
private fun HandlerMethod.hasHttpServletResponseParam() = this.methodParameters
        .any { it.parameterType.name == "javax.servlet.http.HttpServletResponse" }
```
It also gets rid of the nullable code.

Reply:
Well, `HttpServletResponse` is [an interface](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html), so that code snippet can never be true. Moreover, if it was a class, that snippet wouldn't support child classes.